### PR TITLE
Migrate Sticky header

### PR DESCRIPTION
**Pull request by saugatmalla**
* Migrate Sticky header

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/saugat/sticky-header?martech=off
- After: https://sticky-header--milo--adobecom.hlx.page/drafts/saugat/sticky-header?martech=off

### CODE REVIEW

File: libs/blocks/sticky-header/sticky-header.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export default function decorate($block) {
+  const $header = document.querySelector('header');
+  const $div = $block.firstElementChild;
+  const $a = $block.querySelector('a');
+  const $close = createTag('a', { class: 'sticky-header-close' });
+
+  const { miloLibs, codeRoot } = getConfig();
+  const base = miloLibs || codeRoot;
+  const $closeIcon = createTag('img', { class: 'sticky-header-close-icon', src: `${base}/blocks/sticky-header/close.svg` });
```
Comment (vhargrave) [inline libs/blocks/sticky-header/sticky-header.js:11]: The close icon is currently too small on mobile devices. Can you look into that? 
image

File: libs/blocks/sticky-header/sticky-header.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { createTag, getConfig } from '../../utils/utils.js';
+
+export default function decorate($block) {
```
Comment (vhargrave) [inline libs/blocks/sticky-header/sticky-header.js:3]: nit: I don't love having a dollar behind of most of your variables. I can see you're using it to identify html elements, but I think it's unnecessary tbh. It's not a standard that we have and just makes all your variable names longer.

File: libs/blocks/sticky-header/sticky-header.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  const $header = document.querySelector('header');
+  const $div = $block.firstElementChild;
+  const $a = $block.querySelector('a');
```
Comment (vhargrave) [inline libs/blocks/sticky-header/sticky-header.js:6]: what happens if there's a link in the text? You'd then be converting that link into a button.

File: libs/blocks/sticky-header/sticky-header.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  const $close = createTag('a', { class: 'sticky-header-close' });
+
+  const { miloLibs, codeRoot } = getConfig();
+  const base = miloLibs || codeRoot;
+  const $closeIcon = createTag('img', { class: 'sticky-header-close-icon', src: `${base}/blocks/sticky-header/close.svg` });
+
+  $header.parentNode.insertBefore($block, $header.nextElementSibling);
+
+  $a.classList.add('button');
+  $a.target = '_blank';
```
Comment (vhargrave) [inline libs/blocks/sticky-header/sticky-header.js:16]: if you want a link to open in a new tab just add #_blank to the end of the link. This doesn't need to be in the code.

**Review by vhargrave — CHANGES_REQUESTED** [review]
I really like how it looks - especially the button! :) 
I've found a few small things worth looking at . Also please fix the unit tests, some are failing atm.

**Comment by rgclayton** [comment]
I can't help but wonder if creating this in a bit more generic way and creating a couple variants would be more useful. Some common use cases I've seen with the sticky banner/content is sticking to the top _or_ bottom of the page. 

I also think the block name `sticky header` is too verbose. I think something as simple as `sticky`  would be suffice (the sticky block). With variants something like `Sticky (top)` or `Sticky (bottom)`. 

 Thoughts?
 
 cc: @auniverseaway @chrischrischris @ryanmparrish

File: libs/blocks/sticky-header/sticky-header.js (hunk 3 above)
Comment (saugatmalla) [reply libs/blocks/sticky-header/sticky-header.js]: I have made changes to the selector. Please,  let me know if there is a better solution.

File: libs/blocks/sticky-header/sticky-header.js (hunk 4 above)
Comment (saugatmalla) [reply libs/blocks/sticky-header/sticky-header.js]: Interesting point. I wasn't aware of it.

**Review by chrischrischris — CHANGES_REQUESTED** [review]
I agree with @rgclayton that this should be made to work both top and bottom (and renamed to just `sticky`.

Also this doesn't work with section metadata.  I attempted to use a grey background and it is applied after the header:
https://sticky-header--milo--adobecom.hlx.page/drafts/cpeyer/sticky-header-copy

**Comment by saugatmalla** [comment]
@rgclayton @chrischrischris 
Should we keep the sticky bottom block right above the cookie consent message when it is there?
image

**Comment by chrischrischris** [comment]
We're going to close all Draft PRs to keep the PR count down.  If you would like a review of draft code please reach out on #milo-community with a link to your branch.

Thanks!

**Pull request closed without merging**